src/transformers/modeling_utils.py

```diff
@@ -2720,7 +2722,7 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
             model.tie_weights()
             device_map = infer_auto_device_map(model, dtype=torch_dtype if not load_in_8bit else torch.int8, **kwargs)
```

Comment:
Here I would definitely add a check to make sure  `device_map="auto"` would work as expected. One should import the recent `CustomDtype` from `accelerate` here: https://github.com/huggingface/accelerate/pull/1434 
and the logic flow would look like:
1- check if `accelerate` is on the appropriate version and import `CustomDtype` and pass `CustomDtype.int4`
2- if not, log a warning and suggest users to upgrade accelerate
Wdyt?

```python
if load_in_4_bit:
    _accelerate_has_custom_dtype = version.parse(importlib_metadata.version("accelerate")) > version.parse("0.19.0")
    if _accelerate_has_custom_dtype:
        from accelerate.utils import CustomDtype
        target_dtype = CustomDtype.int4
    else:
       logger.warn("You are using `device_map="auto"` on a 4bit loaded version of the model. To compute the appropriate device map, you should upgrade your `accelerate` library, `pip install --upgrade accelerare` or install it from source to support int4 auto device map calculation. You may encounter unexpected behavior, or pass your own device map")
       target_dtype = torch.int8


```